rust/processor/src/models/token_v2_models/v2_token_ownerships.rs

```diff
@@ -80,6 +80,47 @@ pub struct CurrentTokenOwnershipV2 {
     pub non_transferrable_by_owner: Option<bool>,
 }
 
+impl From<CurrentTokenOwnershipV2> for CurrentDeletedTokenOwnershipV2 {
+    fn from(val: CurrentTokenOwnershipV2) -> Self {
+        CurrentDeletedTokenOwnershipV2 {
+            token_data_id: val.token_data_id,
+            property_version_v1: val.property_version_v1,
+            owner_address: val.owner_address,
+            storage_id: val.storage_id,
+            amount: val.amount,
+            table_type_v1: val.table_type_v1,
+            token_properties_mutated_v1: val.token_properties_mutated_v1,
+            is_soulbound_v2: val.is_soulbound_v2,
+            token_standard: val.token_standard,
+            is_fungible_v2: val.is_fungible_v2,
+            last_transaction_version: val.last_transaction_version,
+            last_transaction_timestamp: val.last_transaction_timestamp,
+            non_transferrable_by_owner: val.non_transferrable_by_owner,
+        }
+    }
+}
+
+// because we have the impl on Vec<T> we need to create a wrapper struct to implement the trait if
+// we want to use the same underlying type between different queries
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(token_data_id, property_version_v1, owner_address, storage_id))]
+#[diesel(table_name = current_token_ownerships_v2)]
+pub struct CurrentDeletedTokenOwnershipV2 {
```

Comment:
haha yep i did this originally but removed it because it's not DRY :)

Reply:
now I removed too! nice :-)